test/decl/init/nonnominal_init.swift

```diff
@@ -9,8 +9,10 @@ indirect enum Or<A, B> {
 
 func deMorgan<A, B>(_ ne: Not<Or<A, B>>) -> And<Not<A>, Not<B>> {
   return And<Not<A>, Not<B>>(
-    Not<A> { a in ne(.left(a)) }, // expected-error {{non-nominal type 'Not<A>' (aka '(A) -> Never') does not support explicit initialization}}
-    Not<B> { a in ne(.right(a)) }
+    Not<A> { a in ne(.left(a)) }, // expected-error {{type 'Not<A>' (aka '(A) -> Never') has no member 'init'}}
```

Comment:
It seems to be that the original message here was somewhat better... Or maybe we shouldn't even talk about "initialization" here at all because really `Not` is a function which means that it the problem here is that `trailing closure passed to parameter of type 'A' that does not accept it`? I think this is at least worth a `FIXME(diagnostics)` :)

Reply:
Yes, the original message is better IMO, that particular diagnostic just hasn't been ported yet (and this change causes us to find a solution for this code). I agree though that there should maybe be a more tailored diagnostic because it's still not super clear. I'll add a FIXME

Reply:
Thank you!